File: Golang/abc188_b/main.go
package main

import (
	"fmt"
)

func read_input_list(n int) []int {
	a := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}

	return a
}

func main() {
	var n int
	fmt.Scan(&n)

	a_list := read_input_list(n)
	b_list := read_input_list(n)

	result := solve(a_list, b_list)

	fmt.Println(result)
}

func solve(a_list []int, b_list []int) string {

	sum := 0

	for i := 0; i < len(a_list); i++ {
		sum += a_list[i] * b_list[i]
	}

	if sum == 0 {
		return "Yes"
	}
	return "No"
}

File: Elixir/abc188_b/lib/main.ex
defmodule Main do
  def main do
    _ = InputHelper.read_line()
    a_list = InputHelper.read_integer_list()
    b_list = InputHelper.read_integer_list()

    result = solve(a_list, b_list)
    IO.puts(result)
  end

  def solve(a_list, b_list) do
    Enum.zip(a_list, b_list)
    |> Enum.map(fn {x, y} -> x * y end)
    |> Enum.sum()
    |> then(fn
      0 -> "Yes"
      _ -> "No"
    end)
  end
end

defmodule InputHelper do
  def read_line() do
    IO.read(:line)
  end

  def read_integer() do
    IO.read(:line) |> String.trim() |> String.to_integer()
  end

  def read_integer_list() do
    IO.read(:line)
    |> String.trim()
    |> String.split()
    |> Enum.map(&String.to_integer/1)
  end
end
